reVX/offshore/dist_to_ports.py

```diff
@@ -0,0 +1,395 @@
+# -*- coding: utf-8 -*-
+"""
+Compute least-cost distance to port
+"""
+from concurrent.futures import as_completed
+import geopandas as gpd
+import logging
+import numpy as np
+import os
+import pandas as pd
+from scipy.spatial import cKDTree
+from skimage.graph import MCP_Geometric
+from sklearn.metrics.pairwise import haversine_distances
+
+from reV.handlers.exclusions import ExclusionLayers
+from reVX.utilities.exclusions_converter import ExclusionsConverter
+from reVX.wind_dirs import row_col_indices
+from rex.utilities.execution import SpawnProcessPool
+from rex.utilities.utilities import check_res_file
+
+logger = logging.getLogger(__name__)
+
+
+class DistanceToPorts:
+    """
+    Compute the distance to port in km
```

Comment:
Maybe consider adding more description of the use case in the class docstring. For example, "use this class to compute the least cost distance from the exclusion grid pixels to port locations. The distance to shore layer will be used to calculate least cost paths around land masses and other such obstructions" or something like that?

And also maybe write down somewhere that you can update the least cost exclusion layer by adding in new ports? So like if you get one new port you can just run with a single input port and an existing least cost distnace layer? I think that is really powerful but kind of implicit right now.

Reply:
Can do